python/Ganga/GPIDev/Lib/Job/Job.py

```diff
@@ -1758,6 +1758,8 @@ def doit(f):
                         self.application.decrementShareCounter(self.application.is_prepared.name)
                         for _ in self.subjobs:
                             self.application.decrementShareCounter(self.application.is_prepared.name)
+                    if self.application.getShareCounterVal(self.application.is_prepared.name)==0:
```

Comment:
Would it not be better to implement this inside the SharedDir object itself. It can just do it when its internal counter goes to zero. That would also prevent a potential race condition that exist with this solution with jobs being removed in parallel threads.